### PR TITLE
Method unifications and package movements

### PR DESCRIPTION
**Pull request by ahmetmircik**
ee counterpart https://github.com/hazelcast/hazelcast-enterprise/pull/3487

- Addressing missing parts of @ihsandemir review comment: https://github.com/hazelcast/hazelcast/pull/15555#discussion_r359567850 (deleted `tryReserveForCacheOnUpdate` in favor of `tryReserveForUpdate`, now interface is deeper)
- Aligned test packages based on impl packages

### CODE REVIEW

File: hazelcast/src/main/java/com/hazelcast/internal/nearcache/impl/DefaultNearCache.java [hunk 1]
```diff
@@ -198,18 +202,9 @@ public boolean isPreloadDone() {
 
     @Override
     public long tryReserveForUpdate(K key, Data keyData) {
-        nearCacheRecordStore.doEviction(false);
```
Comment (ihsandemir) [inline hazelcast/src/main/java/com/hazelcast/internal/nearcache/impl/DefaultNearCache.java]: why is this line deleted?

Comment (ahmetmircik) [reply hazelcast/src/main/java/com/hazelcast/internal/nearcache/impl/DefaultNearCache.java]: oops, put it back.
